### PR TITLE
Build BOM file to distribute D4J dependencies

### PR DESCRIPTION
**Pull request by stachu540**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
image

### Changes Proposed in this Pull Request
* creating BOM file to distribute D4J dependecies

### Additional notes
Distribute packages using BOM file will distribute easily dependencies without conflicts with library version. Last time created BOM file, tested it out and results (see image above). Restricting imports jar files for generating only `bom` file to manage distribution (jar files are not required).

Pinging @quanticc to check this PR.

### CODE REVIEW

**Comment by danthonywalker** [comment]
I'd consider `${project.version}` for module versions instead of excessive repeating.

**Comment by stachu540** [comment]
Jitpack won't resolve version correctly so ill stay like this.

File: BOM/build.gradle [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+compileJava.enabled = false
+javadoc.enabled = false
+tasks.withType(Jar) {
+    enabled false
+}
+test.enabled = false
+
+dependencyManagement {
+    dependencies {
+        project.rootProject.subprojects.each {
```
Comment (stachu540) [inline BOM/build.gradle]: Annotate to this [comment](https://github.com/Discord4J/Discord4J/pull/449#issuecomment-423048652)

**Comment by quanticc** [comment]
Do you have an usage example?

We're looking forward to have this feature, but we're not sure if it's going to live as a subproject in _this_ repository, considering we also want to BOM versions on the [Stores](https://github.com/Discord4J/Stores) project.

/cc @Discord4J/members

**Comment by stachu540** [comment]
[My project](https://github.com/GlitchLib/glitch) is a good example.

and [pom file](https://bintray.com/stachu540/GlitchLib/download_file?file_path=io%2Fglitchlib%2Fglitch-BOM%2F0.1.0%2Fglitch-BOM-0.1.0.pom)

I check it and all looks perfect.

**Comment by quanticc** [comment]
Alright thanks, we should just decide the location of it then, either in this repository or under a new one, considering we also want to include Stores project in our BOM.

@Discord4J/members what do you think?

**Comment by stachu540** [comment]
Adding Stores outside `project.rootProject.subprojects.each` block consider your dependency resolves. But be warn, if Stores solution have library conflicts you must resolve it manually editing `BOM\build.gradle`

**Comment by quanticc** [comment]
We've decided to include your proposed changes into a new repository: https://github.com/Discord4J/BOM

**Pull request closed without merging**